### DISCUSSION TITLE
Binary of concatenated string

### DISCUSSION THREAD

**Discussion by ksallberg** [lfe/lfe#457] [Q&A] [answered]
Not sure if i'm missing out something simple, but seems like I cannot make a binary out of a concated string:
`(binary "hejhej") <- OK`

```
lfe> (binary (++ "foo" "bar"))
exception error: #(undefined_bittype 'erlang)
```

**Comment by yurrriq** [comment]
You can do this at least `(list_to_binary (++ "foo" "bar"))`.

**Comment by ksallberg** [comment]
Right :) Thanks! I will leave this ticket open and someone else can decide whether to fix or close

**Comment by deadtrickster** [comment]
What the difference here binary vs list_to_binary

________________________________

**Comment by rvirding** [comment, marked as answer]
`list_to_binary` is a function takes a list of bytes (actually a nested list of bytes and binaries) and builds a binary while `binary` is a special form which builds a binary from a set of segments. These segments can only be in specific formats, for example integers, and not general lists. General lists are not allowed as segments. Check `lfe_guide` for examples of segments and the erlang documentation for all the options. Strings are specially treated and are allowed as segments

I should probably improve the documentation on binaries in `lfe_guide`.
